idl/internal/docstring.go

```diff
@@ -0,0 +1,138 @@
+package internal
+
+import (
+	"strings"
+	"unicode"
+)
+
+const (
+	_docstringOpen  = "/**"
+	_docstringClose = "*/"
+)
+
+// CleanupDocstring takes a docstring in the form,
+//
+//  /**
+//   * foo bar
+//   */
+//
+// And returns,
+//
+//  foo bar
+func CleanupDocstring(s string) string {
+	lines := strings.Split(s, "\n")
+	dedent(lines, true /* skipFirstIfUnindented */)
+
+	switch len(lines) {
+	case 0:
+		return s
+	case 1:
+		// /** foo */
+		s := lines[0]
+		s = strings.TrimPrefix(s, _docstringOpen)
+		s = strings.TrimSuffix(s, _docstringClose)
+		return strings.TrimSpace(s)
+	}
+
+	if len(lines) > 0 {
+		lines[0] = strings.TrimPrefix(lines[0], _docstringOpen)
+		lines[len(lines)-1] = strings.TrimSuffix(lines[len(lines)-1], _docstringClose)
+	}
```

Comment:
this is basically the same logic that the `case 1` also has, trim the `_docstringOpen` and `_docstringClose` from the first and last line. Maybe the logic should look like:
```go
lines := strings.Split(...)
last := len(lines)-1

// Trim the /** and */ from the first and last lines.
lines[0] = strings.TrimPrefix(lines[0], _docstringOpen)
lines[last] = strings.TrimSuffix(lines[last], _docstringClose)

if len(lines) == 1 {
  return strings.TrimSpace(lines[0])
}

lines = dropLeadingEmptyLines(lines)
[...]
```